### DISCUSSION TITLE
Where sumatra PDF stores the "current page" of pdfs

### DISCUSSION THREAD

**Discussion by un-lo-so** [sumatrapdfreader/sumatrapdf#2587] [Q&A]
Hello
I'm a enthusiast user of sumatra pdf reader: I appreciate the lightweight and no-frills interface. A very useful feature, for me, is the capability to remember the exact position of a PDF file when sumatra is closed and reopen the document. My question is: where sumatra store these informations? I ask this because I wish setup my backup program to backup these information, so, if my HD crash abruptly... well... I hope to restore my data... so what files I should include in my backup job? 

I try some test and according to my trial these informations aren't stored in C:\Users\Enrico\AppData\Local\SumatraPDF\SumatraPDF-settings.txt and never embedded in the pdf file.

Thank you!

**Comment by kjk** [comment]
It is in `SumaptraPDF-settings.txt` in `FileStates` section and `SessionsData` section. For example:
```
FileStates [
	[
		FilePath = C:\Users\kjk\Downloads\pragmatism.pdf
		Favorites [
		]
		IsPinned = false
		IsMissing = false
		OpenCount = 6
		UseDefaultState = false
		DisplayMode = continuous
		ScrollPos = -1 69.7494
		PageNo = 123
		Zoom = fit page
		Rotation = 0
		WindowState = 1
		WindowPos = 991 196 1267 1433
		ShowToc = true
		SidebarDx = 312
		DisplayR2L = false
		ReparseIdx = 0
	]
]
```

`PageNo` is where we remember page to restore on re-opening the doc.

Also see https://www.sumatrapdfreader.org/docs/How-we-store-settings

**Reply by un-lo-so** [reply]
Thank you
But I try to modify with notepad the value of PageNo and don't work: If I close sumatra pdf at page 20 and edit the PageNo with value 5 the file will be reopen at page 20. 

I suppose that the information was embedded in the pdf but my backup software don't backup a new file version because the file is the same.

**Reply by kjk** [reply]
Make sure to modify `SessionsData` and `FileStates`

**Reply by kjk** [reply]
And make sure Sumatra is not running as it might over-write the changes.

**Reply by GitHubRulesOK** [reply]
And ensure it is the only copy of SumaptraPDF-settings.txt since you may have used more than one over time

easy way to check is open advanced settings in notepad select `save as` will show the current location you can copy to use AFTER close SumatraPDF

![image](https://user-images.githubusercontent.com/12754913/169376320-e6680c79-d0dc-433b-8a19-2f7ab5df17ed.png)

**Reply by un-lo-so** [reply]
>and `FileStates`
Ok, It works! so in order to restore in a consistent way sumatra I should backup "SumatraPDF-settings.txt" and "sumatrapdfcache" directory too. Correct?

**Reply by GitHubRulesOK** [reply]
The cache is optional as it is only the thumbnails for start screen
so can be easily auto rebuilt by open first 10 files.

Currently I have only 20 in one copy and 18 in another, and there is no easy way to decide which are the first 10 since they use hashed numbers
So if too many over time I usually delete the lot to simply start with fresh 10
